cmd/influxd/launcher/launcher.go

```diff
@@ -970,6 +975,40 @@ func (m *Launcher) initTracing(opts *InfluxdOpts) {
 	}
 }
 
+// writePIDFile will write the process ID to pidFile and register a cleanup function to delete it during
+// shutdown. If pidFile is empty, then no PID file is written and no cleanup function is registered.
+func (m *Launcher) writePIDFile(pidFile string) error {
+	if pidFile == "" {
+		return nil
+	}
+
+	// Create directory to PIDfile if needed.
+	if err := os.MkdirAll(filepath.Dir(pidFile), 0777); err != nil {
+		return fmt.Errorf("mkdir: %w", err)
```

Comment:
The context of the PID filename is added by the caller: https://github.com/influxdata/influxdb/blob/a91ba1cff7900af32b2beb4c3c1b5610ef5fe73c/cmd/influxd/launcher/launcher.go#L252-L254

I figured this was acceptable elimination of code inside `writePIDFile` since it is a private, single-use method. For a public method I definitely would have added the filename into each message.